youtube_dl/YoutubeDL.py

```diff
@@ -1524,7 +1524,7 @@ def dl(name, info):
                         fd.add_progress_hook(ph)
                     if self.params.get('verbose'):
                         self.to_stdout('[debug] Invoking downloader on %r' % info.get('url'))
-                    return fd.download(name, info)
+                    return fd.download(name, info, self.params.get("md5"))
```

Comment:
Calculate it in chunks after the final file is on disk. As said it makes no sense to calculate it immediately since the final file may be modified by postprocessor.